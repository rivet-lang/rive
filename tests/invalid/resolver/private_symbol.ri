mod my_mod {
    struct PrivateStruct {
        a: i32;
        b: PrivateEnum;
    }
    const PRIVATE_CONST: i32 = 1;

    enum PrivateEnum { Variant }
}

mod my_mod2 {
    pub fn func() {
        _ = super::my_mod::PrivateStruct{
            a: super::my_mod::PRIVATE_CONST,
            b: super::my_mod::PrivateEnum::Variant
        };
    }
}

fn main() {
    my_mod2::func();
}
