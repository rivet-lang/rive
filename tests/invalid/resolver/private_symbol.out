tests/failing/resolver/private_symbol.ri:13:28: error: type `PrivateStruct` is private
     |
  13 |         _ = super::my_mod::PrivateStruct{
     |                            ^
tests/failing/resolver/private_symbol.ri:14:31: error: constant `PRIVATE_CONST` is private
     |
  14 |             a: super::my_mod::PRIVATE_CONST,
     |                               ^
tests/failing/resolver/private_symbol.ri:15:31: error: type `PrivateEnum` is private
     |
  15 |             b: super::my_mod::PrivateEnum::Variant
     |                               ^
rivetc: error: could not compile package `core`, aborting due to 3 previous errors